### PR TITLE
Properly dispatch keypress event

### LINKED ISSUE #14659: Github wiki search doesn't work
When visiting https://github.com/servo/servo/wiki in Servo, you can type in the "Find a page" input box, but the links below it never update.

### PR DESCRIPTION
**Pull request by ghost**
This was an attempt to fix #14659. It turned out that the problem wasn't what I thought it was. So, I didn't fix that. On the brighter side, this fixes two related issues.

- Previously, we were unable to launch `keypress` events from `input` and `textarea` elements, because [we'd been cancelling](https://github.com/servo/servo/blob/1327ebd52f53f5f6637a12fab6cf0cad0aa0be6f/components/script/dom/htmlinputelement.rs#L1120-L1124) the key events, so that they don't trigger window navigation - #8400). I've introduced an enum to represent an additional state to an event's cancellation.
- [According to the spec](https://w3c.github.io/uievents/#keypress-event-order), `keypress` (if available) should be dispatched immediately after `keydown`, and it should be followed by `input`. Canceling `keypress` should also cancel `input`. But, we'd been dispatching `input` before `keypress`. We now dispatch `input` once the `keypress` event is on the respective elements.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because it's a refactor?

r? @jdm or anyone interested

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14738)

### CODE REVIEW

File: components/script/dom/event.rs [hunk 1]
```diff
@@ -77,14 +77,21 @@ impl From<bool> for EventCancelable {
     }
 }
 
+#[derive(JSTraceable, HeapSizeOf, Copy, Clone, PartialEq)]
+pub enum EventDefault {
```
Comment (Ms2ger) [inline components/script/dom/event.rs]: This is going to need some clear documentation

Comment (Wafflespeanut) [inline components/script/dom/event.rs]: Certainly :)

**Comment by Wafflespeanut** [comment]
Do we need tests for this?

**Comment by jdm** [comment]
I don't think we can write automated tests for this yet, because we can't synthesize key input.

**Comment by jdm** [comment]
@bors-servo: r+
Nice work. Sorry about the review delay!

**Comment by Wafflespeanut** [comment]
Oh please... I haven't actually fixed the issue yet. I'll get back into it once again this weekend :smile:

**Pull request merged**